solang-parser/src/lexer.rs

```diff
@@ -553,6 +561,9 @@ static KEYWORDS: phf::Map<&'static str, Token> = phf_map! {
     "unchecked" => Token::Unchecked,
     "assembly" => Token::Assembly,
     "let" => Token::Let,
+    "persistent" => Token::Persistent,
+    "temporary" => Token::Temporary,
+    "instance" => Token::Instance,
```

Comment:
This does affect Polkadot/Solana, but never mind, I don't know what other solution there is (other than a better parser generator).

Reply:
Unfortunately the keywords can't be made context-sensitive with the current parser generator. Of course, this is something I would really love and this would be such a nicer feature (for lalrpop).

Having said that, it might be possible to replace the keywords `persistent`, `temporary`, and `instance` with `SolIdentifier` and then checking for the keywords in sema.

Reply:
A way around this is to make these keywords Soroban specific, so they do not apply for Solana/Polkadot. This could be implemented in the lexer.